PennMobile/GSR-Booking/Views/GSRLocationCell.swift

```diff
@@ -26,15 +28,60 @@ struct GSRLocationCell: View {
                 .frame(width: 120, height: 80)
                 .scaledToFill()
                 .clipShape(RoundedRectangle(cornerRadius: 8))
-            Text(location.name)
-                .font(.system(size: 18))
-                .padding(.leading, 16)
+            VStack (alignment: .leading, spacing: 6){
+                Text(location.name)
+                    .font(.system(size: 18))
+                if let count = availabilityCount {
+                    if (count == 0) {
+                        HStack {
+                            Image(systemName:"circle.fill")
+                            Text("All reserved")
+                        }
+                        .font(.system(size: 12))
+                        .foregroundColor(.baseRed)
+                        .padding(.horizontal, 8)
+                        .padding(.vertical, 4)
+                        .background(
+                            RoundedRectangle(cornerRadius: 8)
+                                .fill(.ultraThinMaterial)
+                        )
+                    } else {
+                        HStack {
+                            Image(systemName:"circle.fill")
+                            Text("\(count) open now")
+                        }
+                        .font(.system(size: 12))
+                        .padding(.horizontal, 8)
+                        .padding(.vertical, 4)
+                        .foregroundColor(.baseGreen)
+                        .background(
+                            RoundedRectangle(cornerRadius: 8)
+                                .fill(.ultraThinMaterial)
+                        )
+                    }
+                } else {
+                    ProgressView()
+                }
+            }
+            .padding(.leading, 16)
             Spacer()
             Image(systemName: "chevron.right")
                 .bold()
         }
         .frame(height: height)
         .cornerRadius(8)
         .contentShape(.rect)
+        .task {
+            await loadAvailability()
```

Comment:
I would look for a place to cache this so that the availability doesn't reload every time a location is scrolled into a view. Of course you'd want to make sure the cache isn't stale (like maybe refresh every minute or so) but that's a separate problem